R/utils_render_table_venues.R

```diff
@@ -18,8 +18,20 @@ render_tables_venues_json <- function(list_venue_reg_tables){
 #' @return Returns a JSON table for the all venues 
 render_table_all_venues_json <- function(list_venue_reg_tables){
   col_names <- CONFIG$NON_REG_TABLE_COL_NAMES[["venues_table"]]
-  list_venue_names <- names(list_venue_reg_tables)
-  
+
+  # Since list_venue_reg_tables is of the form list("subcat_1" = list(venue_1), "subcat_2" = list(venue_2))
+  # we need to unpack the list to obtain the venue names
+  list_venue_types <- list()
+  list_venue_names <- list()
+
+  # Unpack the list to obtain the venue names and their corresponding types
+  for (venue_type in names(list_venue_reg_tables)) {
```

Comment:
I cannot compile this in my head right now, but have you checked if the function `unlist` can help you here?

https://www.r-bloggers.com/2023/08/the-unlist-function-in-r/